src/platform/plugins/shared/controls/public/controls/esql_control/get_esql_control_factory.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { BehaviorSubject } from 'rxjs';
+import { css } from '@emotion/react';
+import { EuiComboBox } from '@elastic/eui';
+import { apiPublishesESQLVariables } from '@kbn/esql-variables-types';
+import { useBatchedPublishingSubjects } from '@kbn/presentation-publishing';
+import { ESQL_CONTROL } from '../../../common';
+import type { ESQLControlState, ESQLControlApi } from './types';
+import { ControlFactory } from '../types';
+import { uiActionsService } from '../../services/kibana_services';
+import { initializeDefaultControlApi } from '../initialize_default_control_api';
+import { initializeESQLControlSelections } from './esql_control_selections';
+
+const displayName = i18n.translate('controls.esqlValuesControl.displayName', {
+  defaultMessage: 'Static values list',
+});
+
+export const getESQLControlFactory = (): ControlFactory<ESQLControlState, ESQLControlApi> => {
+  return {
+    type: ESQL_CONTROL,
+    order: 3,
+    getIconType: () => 'editorChecklist',
+    getDisplayName: () => displayName,
+    buildControl: async (initialState, buildApi, uuid, controlGroupApi) => {
+      const defaultControl = initializeDefaultControlApi(initialState);
+      const selections = initializeESQLControlSelections(initialState);
+
+      const onSaveControl = (updatedState: ESQLControlState) => {
+        controlGroupApi?.replacePanel(uuid, {
+          panelType: 'esqlControl',
+          initialState: updatedState,
+        });
+      };
+
+      const api = buildApi(
+        {
+          ...defaultControl.api,
+          ...selections.api,
+          defaultPanelTitle: new BehaviorSubject<string | undefined>(initialState.title),
```

Comment:
FYI, https://github.com/elastic/kibana/pull/205279 has merged, changing `defaultPanelTitle` to `defaultTitle`. Please be sure to rebase with main and update api's to there new naming conventions.

Reply:
Thanx for the heads up, done 👍